### PR TITLE
Add failure tests for noms-merge

### PR DESCRIPTION
**Pull request by cmasone-attic**
Tests command line validation for noms-merge

Toward #2445

### CODE REVIEW

**Comment by cmasone-attic** [comment]
ptal

**Comment by cmasone-attic** [comment]
Ping?

File: samples/go/noms-merge/main.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ func nomsMerge() error {
 		}
 
 		d.PanicIfTrue(flag.NArg() != 3, "Incorrect number of arguments\n")
+		d.PanicIfTrue(*parentStr == "", "--parent is currently required\n")
```
Comment (kalman) [inline samples/go/noms-merge/main.go]: I'd drop the "currently"

File: samples/go/noms-merge/main_test.go [hunk 2]
```diff
@@ -78,4 +79,41 @@ func (s *testSuite) TestWin() {
...
+	prep(r)
+	db.Close()
+
+	for _, c := range cases {
+		var mainErr error
+		stdout, _ := s.Run(func() { mainErr = nomsMerge() }, c.args)
+		s.Empty(stdout, "Expected empty stdout for case: %#v", c.args)
+		if s.Error(mainErr) {
+			s.Equal(c.err, mainErr.Error(), "Unexpected output for case: %#v\n", c.args)
+		}
```
Comment (kalman) [inline samples/go/noms-merge/main_test.go]: Do you want to fail if there isn't an error?

Comment (kalman) [inline samples/go/noms-merge/main_test.go]: I.e.
```
var mainErrStr string
if s.Error(mainErr) {
  mainErrStr = mainErr.Error()
}
s.Equal(c.err, mainErrStr, "unexpected output for case...")
```

**Comment by kalman** [comment]
lgtm with comments

File: samples/go/noms-merge/main.go (hunk 1 above)
Comment (cmasone-attic) [inline samples/go/noms-merge/main.go]: dun

File: samples/go/noms-merge/main_test.go (hunk 2 above)
Comment (cmasone-attic) [inline samples/go/noms-merge/main_test.go]: this will fail if there isn't an error. `s.Error()` will fail, causing the test case to fail

**Pull request merged**